pip/commands/install.py

```diff
@@ -369,3 +359,33 @@ def run(self, options, args):
                 )
             shutil.rmtree(temp_target_dir)
         return requirement_set
+
+
+class InstallCommand(InstallBase):
+    """
+    Install packages from:
+
+    - PyPI (and other indexes) using requirement specifiers.
+    - VCS project urls.
+    - Local project directories.
+    - Local or remote source archives.
+
+    pip also supports installing from "requirements files", which provide
+    an easy way to specify a whole environment to be installed.
+    """
+    name = 'install'
+    summary = 'Install packages.'
+    upgrade_command = False
+    upgrade_option = True
+    upgrade_is_recursive = True
```

Comment:
you've got an inconsistency here.   `InstallCommand` has `upgrade_is_recursive`, whereas  `UpgradeCommand` has `upgrade_recursive`.

I'd prefer the shorter  `upgrade_recursive`